### PR TITLE
Run3-gem62C Use ESGetToken in RPC geometry packages

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Use ESGetToken in RPC geometry packages

#### PR validation:

Use runTheMatrix test workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/RPCGeometryBuilder/plugins/RPCGeometryValidate.cc [hunk 1]
```diff
@@ -101,16 +101,17 @@ class RPCGeometryValidate : public one::EDAnalyzer<> {
...
       tolerance_(iConfig.getUntrackedParameter<int>("tolerance", 6)) {
   fwGeometry_.loadMap(infileName_.c_str());
   outFile_ = new TFile(outfileName_.c_str(), "RECREATE");
 }
 
 void RPCGeometryValidate::analyze(const edm::Event& event, const edm::EventSetup& eventSetup) {
-  eventSetup.get<MuonGeometryRecord>().get(rpcGeometry_);
-  if (rpcGeometry_.isValid()) {
+  rpcGeometry_ = &eventSetup.getData(tokRPC_);
+  if (rpcGeometry_ != nullptr) {
```
Comment (cvuosalo) [inline Geometry/RPCGeometryBuilder/plugins/RPCGeometryValidate.cc:114]: These lines change functionality. See this comment:
https://github.com/cms-sw/cmssw/pull/34214#discussion_r656695574

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**